src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnvDTE;
+using NuGet.Configuration;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+using NuGet.ProjectModel;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(IVsPathContextProvider))]
+    public sealed class VsPathContextProvider : IVsPathContextProvider
+    {
+        private readonly Lazy<ISettings> _settings;
+        private readonly Lazy<IVsSolutionManager> _solutionManager;
+        private readonly Func<BuildIntegratedNuGetProject, Task<LockFile>> _getLockFileOrNullAsync;
+
+        [ImportingConstructor]
+        public VsPathContextProvider(
+            Lazy<ISettings> settings,
+            Lazy<IVsSolutionManager> solutionManager)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            if (solutionManager == null)
+            {
+                throw new ArgumentNullException(nameof(solutionManager));
+            }
+
+            _settings = settings;
+            _solutionManager = solutionManager;
+            _getLockFileOrNullAsync = BuildIntegratedProjectUtility.GetLockFileOrNull;
+        }
+
+        /// <summary>
+        /// This constructor is just used for testing.
+        /// </summary>
+        public VsPathContextProvider(
+            ISettings settings,
+            IVsSolutionManager solutionManager,
+            Func<BuildIntegratedNuGetProject, Task<LockFile>> getLockFileOrNullAsync = null)
```

Comment:
It shouldn't have a default value. It'd resolve to the first one if a value is not provided.